server/backend/neighbors.py

```diff
@@ -172,39 +196,38 @@ def query_model_for_tok(
     result = []
     word_vectors = model if use_keyedvec else model.wv
 
-    cutoff_index = min(
-        map(
-            lambda x: (
-                999999
-                if word_vectors.get_vecattr(x[1], "count") > word_freq_count_cutoff
-                else x[0]
-            ),
-            enumerate(word_vectors.index_to_key),
-        )
-    )
-
     # Check to see if token is in the vocab
     vocab = set(word_vectors.key_to_index.keys())
 
     if tok in vocab:
         # If it is grab the neighbors
         # Gensim needs to be > 4.0 as they enabled neighbor clipping (remove words from entire vocab)
-        word_neighbors = word_vectors.most_similar(
-            tok,
-            topn=neighbors,
-            clip_end=cutoff_index,
-        )
+        word_neighbors = word_vectors.most_similar(tok, topn=neighbors)
 
         # Append neighbor to word_neighbor_map
         for neighbor in word_neighbors:
-            result.append(neighbor[0])
+            word_neighbor = neighbor[0]
+            is_tagged = False
+
+            # Convert tags that contain the following pattern
+            # disease_mesh_####### or chemical_mesh_#######
+            if "mesh" in word_neighbor:
+                word_neighbor = "_".join(word_neighbor.split("_")[1:])
+
+            # Insert tagged suffix to show users that
+            # some concpets are tagged and some concepts are missed
+            # example: mcf-7 is a cellline but the token itself appears as well
+            if word_neighbor["token"] in concept_id_mapper_dict:
+                word_neighbor = concept_id_mapper_dict[word_neighbor]
+                is_tagged = True
```

Comment:
Also, it occurred to me that perhaps rather than simply returning an `is_tagged` boolean, we could return the ID prior to remapping, in case someone who consumes the API wants that information.

Reply:
So have a field that provides the tag's id if present otherwise empty string?

Reply:
Exactly, yeah; well, I prefer using `None` (in JSON, it'll end up as `null`) rather than an empty string. Maybe we change `is_tagged` to `tag_id` in the token dict, then set that field to `None` unless it was tagged?